include/util/struct_packing.hpp

```diff
@@ -0,0 +1,19 @@
+#ifndef STRUCT_PACKING_HPP
+#define STRUCT_PACKING_HPP
+
+/*
+    Macro to pack struct/class in both environment
+    Usage:  PACK( 
+        struct MyStruct 
+        { 
+            int Property1;
+        });
+*/
+
+#ifdef _MSC_VER
+#define PACK( input ) __pragma( pack(push, 1) ) input __pragma( pack(pop) )
```

Comment:
@oxidase Thanks man, I think I understand now after catching up on some reading about SIGBUS and ubsan.  Tell you the truth I'm a .net guy and haven't touch c++ long time and no gcc either, so this is a good thing to know.  So I use your coliru linked and compare between #pragma pack vs. attribute, turn on verbosity, I don't see anything different in the compiler message between them and it's silent (http://coliru.stacked-crooked.com/a/9682ca9a0e1bb197).  And both trigger the ubsan with the -fsanitize=undefined just like your original link.

Now back to the pull request, my intention is to fix windows build so I can use it in my project, it seems to me now the question is beyond my PR, so should we do pack to begin with ?